src/main/java/bgu/spl/mics/SimplePublisher.java

```diff
@@ -34,6 +34,7 @@
      * @param b The broadcast message to send
      */
     public final void sendBroadcast(Broadcast b) {
-        //TODO: implement this.
+        MessageBroker m = MessageBrokerImpl.getInstance();
+        m.sendBroadcast(b);
```

Comment:
you can save the calls to getInstance if you'll create a field.. I think it is better because this code is going to be called *a lot*

Reply:
We aren't supposed to add a constructor to SimplePublisher as hinted [here](https://www.cs.bgu.ac.il/~spl201/index.php?page=Assignments.Assignment_2Forum&action=show-thread&id=6c5a320268136354ea97bd10ab945da3).
I will upload an update now that will be more efficient in my opinion

Reply:
you can do it in the initialization